File: Difficulty: Easy/Right View of Binary Tree/right-view-of-binary-tree.cpp
/*
class Node {
  public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = nullptr;
        right = nullptr;
    }
};
*/

class Solution {
  public:
  
    void solve(Node* root, int level, vector<int>& ans) {
        
        if(root==nullptr) return;
        
        if(level==ans.size()) ans.push_back(root->data);
        
        solve(root->right,level+1,ans);
        solve(root->left,level+1,ans);
        
    }
    vector<int> rightView(Node *root) {
        //  code here
        vector<int> ans;
        
        solve(root,0,ans);
        
        
        return ans;
    }
};

File: Difficulty: Easy/Right View of Binary Tree/right-view-of-binary-tree.java
/*
Definition for Node
class Node {
    int data;
    Node left, right;

    Node(int val) {
        this.data = val;
        this.left = null;
        this.right = null;
    }
}
*/

class Solution {
    private void solve(Node root, ArrayList<Integer> ans, int level) {
        if(root==null) return;
        
        if(level==ans.size()) ans.add(root.data);
        
        solve(root.right,ans,level+1);
        solve(root.left,ans,level+1);
    }
    public ArrayList<Integer> rightView(Node root) {
        // code here
        ArrayList<Integer> ans = new ArrayList<>();
        int level = 0;
        solve(root,ans,level);
        
        return ans;
    }
}